### ISSUE TITLE
Encrypt/Decrypt cookies

### ISSUE THREAD

**Issue by oscarotero** [middlewares/ideas#13]
A middleware to decrypt incoming and encrypt outgoing cookies.
It could use [a JWT implementation](https://github.com/lcobucci/jwt/) or [php-encription](https://github.com/defuse/php-encryption).

Maybe it could provide some security features:
* Filter allowed cookies (and remove the others)
* Ensure all cookies have directives like secure, httponly, etc...

**Comment by sa-tasche** [comment]
Why encrypting cookies? HTTPS should be used for that case.
Why filtering? They can be ignored.

**Comment by oscarotero** [comment]
Sometimes you don't want the client be able to read a cookie value, or you cannot use https, etc.
And on remove cookies, they are not included in the next http request, so it's good for performance.
Anyway, these are only random ideas. I'm agree that, in most cases, the need for encrypted cookies is surely because there're other bad things in the app security architecture.

**Comment by ncou** [comment]
Hi,

I made a middleware like this, using a custom crypt engine (https://github.com/ncou/CryptEngine) i could share the code if you want, but i need to tidy it a bit (by the way the credits goes to another person because i was inspired by some piece of code i found on internet).

**Comment by oscarotero** [comment]
Thanks, @ncou 
I was thinking in using a dependency like [php-encryption](https://github.com/defuse/php-encryption) that uses best practices and is actively maintaned, instead implement directly the crypt engine, to do not reinvent the wheel an avoid possible future security issues.

**Comment by ncou** [comment]
yes you are absolutly right this lib should be more stronger and more actively maintained than mine.

**Comment by filisko** [comment]
I'll probably need a middleware like this for a future project. Any ideas you want to push? (goes for everyone)
